tool/ci/pkg/bot/check.go

```diff
@@ -0,0 +1,252 @@
+package bot
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/gravitational/teleport/tool/ci"
+	"github.com/gravitational/teleport/tool/ci/pkg/environment"
+
+	"github.com/google/go-github/v37/github"
+	"github.com/gravitational/trace"
+)
+
+// Check checks if all the reviewers have approved the pull request in the current context.
+func (c *Bot) Check(ctx context.Context) error {
+	pr := c.Environment.Metadata
+	// Remove any stale workflow runs. As only the current workflow run should
+	// be shown because it is the workflow that reflects the correct state of the pull.
+	//
+	// Note: This is run for all workflow runs triggered by an event from an internal contributor,
+	// but has to be run again in cron workflow because workflows triggered by external contributors do not
+	// grant the Github actions token the correct permissions to dismiss workflow runs.
+	if c.Environment.IsInternal(pr.Author) {
```

Comment:
Yes it will leave the latest run which is the run that would be currently running when dismissing the previous workflows. It will not leave an "extra" stale run, it will leave the one running in the current context.